lib/ansible/modules/cloud/amazon/ec2_vpc_route_table_facts.py

```diff
@@ -51,79 +56,175 @@
+              type: string
+        tags:
+          description: Any tags assigned to the VPN connection.
+          returned: always
+          type: dict
+          sample: {
+                      "Name": "test-conn"
+                  }
+        vpc_id:
+          description: The ID of the VPC peering connection.
+          returned: when vpc peering available
+          type: string
+          sample: vpc-a01106c2
+
+'''
+import json
 
 try:
-    import boto.vpc
-    from boto.exception import BotoServerError
-    HAS_BOTO = True
+    from botocore.exceptions import ClientError, BotoCoreError
 except ImportError:
-    HAS_BOTO = False
-
-from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import AnsibleAWSError, connect_to_aws, ec2_argument_spec, get_aws_connection_info
-
+    pass  # caught by imported HAS_BOTO3
 
-def get_route_table_info(route_table):
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (ec2_argument_spec, boto3_conn, ansible_dict_to_boto3_filter_list, HAS_BOTO3,
+                                      camel_dict_to_snake_dict, boto3_tag_list_to_ansible_dict)
 
-    # Add any routes to array
-    routes = []
-    associations = []
-    for route in route_table.routes:
-        routes.append(route.__dict__)
-    for association in route_table.associations:
-        associations.append(association.__dict__)
 
-    route_table_info = {'id': route_table.id,
-                        'routes': routes,
-                        'associations': associations,
-                        'tags': route_table.tags,
-                        'vpc_id': route_table.vpc_id
-                        }
+def date_handler(obj):
+    return obj.isoformat() if hasattr(obj, 'isoformat') else obj
 
-    return route_table_info
 
+def list_route_tables(route_table, module):
+    params = dict()
 
-def list_ec2_vpc_route_tables(connection, module):
-
-    filters = module.params.get("filters")
-    route_table_dict_array = []
+    params['Filters'] = ansible_dict_to_boto3_filter_list(module.params.get('filters'))
+    params['RouteTableIds'] = module.params.get('route_table_ids')
 
     try:
-        all_route_tables = connection.get_all_route_tables(filters=filters)
-    except BotoServerError as e:
-        module.fail_json(msg=e.message)
-
-    for route_table in all_route_tables:
-        route_table_dict_array.append(get_route_table_info(route_table))
-
-    module.exit_json(route_tables=route_table_dict_array)
+        result = json.loads(json.dumps(route_table.describe_route_tables(**params), default=date_handler))
+    except ValueError as e:
+        module.fail_json_aws(e, msg="Cannot validate JSON data")
+    except (ClientError, BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Could not describe route tables")
+    snaked_route_tables = [camel_dict_to_snake_dict(route_table) for route_table in result['RouteTables']]
```

Comment:
I think you might need `camel_dict_to_snake_dict(route_table, ignore_list=['Tags'])`. Looks like CamelCase tags might be modified right now.